src/system/rtos_hooks.c

```diff
@@ -29,11 +29,15 @@ void FIQInterrupt() {
 // Replacement for DataAbortInterrupt
 void DataAbortInterrupt() {
 	taskDISABLE_INTERRUPTS();
+	taskENTER_CRITICAL();
 
 	register int sp;
 	asm("add %0,sp,#8\n" : "=r"(sp));
-	extern void report_data_abort(uint32_t);
-	report_data_abort(sp);
+	for(uint8_t i = 1; i <= 21; i++) {
```

Comment:
This for loop should be moved to after the call to `report_fatal_error`. Currently, it messes up the program counter when printing the stack trace.

Reply:
Fixed in https://github.com/purduesigbots/pros/pull/651/commits/b8f302634494d455f5c0c74918423112030dcb65.